packages/plugin-core/src/commands/MergeNoteCommand.ts

```diff
@@ -0,0 +1,453 @@
+        async (prev, linkToUpdate) => {
+          const acc = await prev;
+          const oldLink = LinkUtils.dlink2DNoteLink(linkToUpdate);
+          const notesWithSameName = await this.extension.getEngine().findNotes({
+            fname: destNote.fname,
+          });
+          const isXVault = oldLink.data.xvault || notesWithSameName.length > 1;
+          const newLink = {
+            ...oldLink,
+            from: {
+              ...oldLink.from,
+              alias:
+                oldLink.from.alias === oldLink.from.fname
+                  ? destNote.fname
+                  : oldLink.from.alias,
+              fname: destNote.fname,
+              vaultName: VaultUtils.getName(destNote.vault),
+            },
+            data: {
+              ...oldLink.data,
+              xvault: isXVault,
+            },
+          };
+          const newBody = LinkUtils.updateLink({
+            note: acc,
+            oldLink,
+            newLink,
+          });
+          acc.body = newBody;
+          return acc;
+        },
+        Promise.resolve(noteToUpdateDeepCopy)
+      );
+
+      noteToUpdate.body = modifiedNote.body;
+      const writeResp = await this.extension
+        .getEngine()
+        .writeNote(noteToUpdate);
+      if (writeResp.data) {
+        return writeResp.data;
+      } else {
+        // We specifically filtered for notes that do have some links to update,
+        // so this is very unlikely to be reached.
+        // Gracefully handle and log error
+        this.L.error({ ctx, message: "No links found to update" });
+        return [];
+      }
+    }
+    // Note to update wasn't found
+    // this will likely never happen given a sound engine state.
+    // Log this as a canary for the engine state, and gracefully return.
+    this.L.error({ ctx, message: "No note found" });
+    return [];
+  }
+
+  /**
+   * Given a source note and dest note,
+   * Look at all the backlinks source note has, and update them
+   * to point to the dest note.
+   * @param sourceNote Source note
+   * @param destNote Dest note
+   */
+  private async updateBacklinks(opts: {
+    sourceNote: NoteProps;
+    destNote: NoteProps;
+  }) {
+    const ctx = "MergeNoteCommand:updateBacklinks";
+    const { sourceNote, destNote } = opts;
+
+    // grab all backlinks from source note
+    const sourceBacklinks = sourceNote.links.filter((link) => {
+      return link.type === "backlink";
+    });
+
+    // scrub through the backlinks and all notes that need to be updated
+    const noteIDsToUpdate = Array.from(
+      new Set(
+        sourceBacklinks
+          .map((backlink) => backlink.from.id)
+          .filter((ent): ent is string => ent !== undefined)
+      )
+    );
+
+    // for each note that needs to be updated,
+    // find all links that need to be updated from end to front.
+    // then update them.
+    let noteChangeEntries: NoteChangeEntry[] = [];
+    await asyncLoopOneAtATime(noteIDsToUpdate, async (id) => {
```

Comment:
does this need to be done sequentially vs in parallel?

Reply:
doesn't need to be sequential. good catch!